task02/queue.cpp

```diff
@@ -0,0 +1,120 @@
+//
+// Created by jacknondaniels on 21.03.19.
+//
+#include <iostream>
+#include "queue.h"
+
+template <typename T>  class Note
+{
+public:
+    void setnext (Note *ptr) {next = ptr;}
+    Note *getnext () {return next;}
+    T &getdata () {return body;}
+    Note (T a = 0):
+            body (a),
+            next (nullptr)
+    {};
+private:
+    T body;
+    Note * next;
+};
+
+template <typename T>  class ListQueue: public BasicQueue <T>
+{
+public:
+    ListQueue ():
+            head (nullptr),
+            size (0)
+    {};
+
+    ~ListQueue() final
+    {if (head) {
+            Note<T> *current;
+            while (head->getnext())
+            {
+                current = head;
+                head = current->getnext();
+                delete current;
+            };
+            delete head;
+        }
+    };
+
+    void enqueue (T&a) override  {
+        if (head)
+        {
+            Note<T> *current = head;
+            while (current->getnext())
```

Comment:
This is OK for little homework, but for real Queue, it is so slow. It is O(n) complexity.
Better to have a pointer to another end of the list. It will have O(1).